### PR TITLE
chore: fix ci

### LINKED ISSUE #6393: Broken CI
Looks like CI is currently broken due to a recent `serde` update; the [advice](https://github.com/serde-rs/serde/issues/2770#issuecomment-2212162225) is to use a more recently nightly.

I see that #6391 is in progress to address this.

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---

Motivation and Context
---

How Has This Been Tested?
---

What process can a PR reviewer use to test or verify this change?
---

Breaking Changes
---

- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -70,6 +70,9 @@ jobs:
             tari-${{ runner.os }}-${{ runner.cpu-model }}-${{ env.toolchain }}-nightly
       - name: cargo format
         run: cargo fmt --all -- --check
+      - run: rustup update
+      - run: rustup default nightly-2024-07-07
+      - run: rustup update
```
Comment (leet4tari) [inline .github/workflows/ci.yml]: As this is a hack, I would rather suggest replacing the above snippet with the following
```yaml
      - name: rust update hack
         run: |
           rustup update
           rustup default ${{ env.toolchain }}
           rustup update
```

File: base_layer/core/src/common/mod.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@ use crate::consensus::DomainSeparatedConsensusHasher;
 
 pub mod borsh;
 pub mod byte_counter;
-pub mod limited_reader;
+// pub mod limited_reader;
```
Comment (AaronFeickert) [inline base_layer/core/src/common/mod.rs]: Would suggest either removing altogether or adding a quick note for why it's kept under the comment.

File: base_layer/core/Cargo.toml [hunk 3]
```diff
@@ -113,3 +113,6 @@ tari_features = { path = "../../common/tari_features", version = "1.0.0-pre.15"
 [[bench]]
 name = "mempool"
 harness = false
+
+[lints.rust]
+unexpected_cfgs = { level = "warn", check-cfg = ['cfg(tari_target_network_mainnet)', 'cfg(tari_target_network_nextnet)', 'cfg(tari_target_network_testnet)'] }
```
Comment (AaronFeickert) [inline base_layer/core/Cargo.toml]: Probably want to add a newline to make this happy.

File: base_layer/tari_mining_helper_ffi/Cargo.toml [hunk 4]
```diff
@@ -30,3 +30,6 @@ tari_common = { path = "../../common", features = ["build", "static-application-
 
 [lib]
 crate-type = ["cdylib"]
+
+[lints.rust]
+unexpected_cfgs = { level = "warn", check-cfg = ['cfg(tari_target_network_mainnet)', 'cfg(tari_target_network_nextnet)', 'cfg(tari_target_network_testnet)'] }
```
Comment (AaronFeickert) [inline base_layer/tari_mining_helper_ffi/Cargo.toml]: Probably want to add a newline to make this happy.

**Review by AaronFeickert — CHANGES_REQUESTED** [review]
There are a few config files and scripts that still reference the old nightly. Would recommend a quick find/replace on those.

Otherwise, just a few nits. I assume none of the removed dead code was a false positive.

**Review by AaronFeickert — APPROVED** [review]
LGTM

**Pull request merged**